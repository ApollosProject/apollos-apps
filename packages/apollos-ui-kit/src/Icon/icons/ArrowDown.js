import React from 'react';
import PropTypes from 'prop-types';
import Svg, { Path } from 'react-native-svg';

import makeIcon from './makeIcon';

const Icon = makeIcon(({ size = 32, fill, ...otherProps } = {}) => (
  <Svg width={size} height={size} viewBox="0 0 24 24" {...otherProps}>
    <Path
      d="M12 18c-.16 0-.32-.06-.46-.2L3 9.66 3.9 8l8.1 7.72L20.1 8l.9 1.65-8.54 8.16c-.14.15-.3.2-.46.2z"
      fill={fill}
    />
  </Svg>
));

Icon.propTypes = {
  size: PropTypes.number,
  fill: PropTypes.string,
};

export default Icon;
