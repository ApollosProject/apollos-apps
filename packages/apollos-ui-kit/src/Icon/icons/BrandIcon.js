import React from 'react';
import PropTypes from 'prop-types';
import Svg, { Path } from 'react-native-svg';

import makeIcon from './makeIcon';

const Icon = makeIcon(({ size = 32, fill, ...otherProps } = {}) => (
  <Svg width={size} height={size} viewBox="0 0 24 24" {...otherProps}>
    <Path
      d="M8.84799129,0 L13.7577354,0 L22.6057267,24 L0,24 L8.84799129,0 Z M6.52559608,19.4398694 L16.0921502,19.4398694 L11.4494502,5.83629831 L11.1682961,5.83629831 L6.52559608,19.4398694 Z"
      fill={fill}
    />
  </Svg>
));

Icon.propTypes = {
  size: PropTypes.number,
  fill: PropTypes.string,
};

export default Icon;
