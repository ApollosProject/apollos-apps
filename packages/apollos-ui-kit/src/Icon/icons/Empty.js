import React from 'react';
import PropTypes from 'prop-types';
import Svg from 'react-native-svg';

import makeIcon from './makeIcon';

const Icon = makeIcon(({ size = 32, ...otherProps } = {}) => (
  <Svg width={size} height={size} viewBox="0 0 24 24" {...otherProps} />
));

Icon.propTypes = {
  size: PropTypes.number,
  fill: PropTypes.string,
};

export default Icon;
