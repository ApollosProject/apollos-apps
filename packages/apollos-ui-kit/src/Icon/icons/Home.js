import React from 'react';
import PropTypes from 'prop-types';
import Svg, { Path } from 'react-native-svg';

import makeIcon from './makeIcon';

const Icon = makeIcon(({ size = 32, fill, ...otherProps } = {}) => (
  <Svg width={size} height={size} viewBox="0 0 24 24" {...otherProps}>
    <Path
      d="M19,2.5 L16.5,2.5 C15.9477153,2.5 15.5,2.05228475 15.5,1.5 C15.5,0.94771525 15.9477153,0.5 16.5,0.5 L20,0.5 C20.5522847,0.5 21,0.94771525 21,1.5 L21,5 C21,5.55228475 20.5522847,6 20,6 C19.4477153,6 19,5.55228475 19,5 L19,2.5 Z M21,22 C21,22.5522847 20.5522847,23 20,23 L14,23 C13.4477153,23 13,22.5522847 13,22 L13,17 L10,17 L10,22 C10,22.5522847 9.55228475,23 9,23 L4,23 C3.44771525,23 3,22.5522847 3,22 L3,13 C3,12.4477153 3.44771525,12 4,12 C4.55228475,12 5,12.4477153 5,13 L5,21 L8,21 L8,16 C8,15.4477153 8.44771525,15 9,15 L14,15 C14.5522847,15 15,15.4477153 15,16 L15,21 L19,21 L19,13 C19,12.4477153 19.4477153,12 20,12 C20.5522847,12 21,12.4477153 21,13 L21,22 Z M12.7071068,0.292893219 L23.7071068,11.2928932 C24.0976311,11.6834175 24.0976311,12.3165825 23.7071068,12.7071068 C23.3165825,13.0976311 22.6834175,13.0976311 22.2928932,12.7071068 L12,2.41421356 L1.70710678,12.7071068 C1.31658249,13.0976311 0.683417511,13.0976311 0.292893219,12.7071068 C-0.0976310729,12.3165825 -0.0976310729,11.6834175 0.292893219,11.2928932 L11.2928932,0.292893219 C11.6834175,-0.0976310729 12.3165825,-0.0976310729 12.7071068,0.292893219 Z"
      fill={fill}
    />
  </Svg>
));

Icon.propTypes = {
  size: PropTypes.number,
  fill: PropTypes.string,
};

export default Icon;
