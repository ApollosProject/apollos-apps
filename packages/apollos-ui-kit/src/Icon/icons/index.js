export ArrowBack from './ArrowBack';
export ArrowDown from './ArrowDown';
export ArrowNext from './ArrowNext';
export ArrowUp from './ArrowUp';
export Audio from './Audio';
export Avatar from './Avatar';
export Bank from './Bank';
export BrandIcon from './BrandIcon';
export Building from './Building';
export Calendar from './Calendar';
export Camera from './Camera';
export Check from './Check';
export CheckboxChecked from './CheckboxChecked';
export CheckboxUnchecked from './CheckboxUnchecked';
export CircleOutlineCheckMark from './CircleOutlineCheckMark';
export CircleOutlineXMark from './CircleOutlineXMark';
export Close from './Close';
export Credit from './Credit';
export Chromecast from './Chromecast';
export Download from './Download';
export Dropdown from './Dropdown';
export Empty from './Empty';
export Failed from './Failed';
export Filter from './Filter';
export Fullscreen from './Fullscreen';
export Groups from './Groups';
export Home from './Home';
export Information from './Information';
export Like from './Like';
export LikeSolid from './LikeSolid';
export Locate from './Locate';
export Lock from './Lock';
export Mute from './Mute';
export Pause from './Pause';
export Pin from './Pin';
export Play from './Play';
export Print from './Print';
export Profile from './Profile';
export RadioChecked from './RadioChecked';
export RadioUnchecked from './RadioUnchecked';
export Repeat from './Repeat';
export RepeatAll from './RepeatAll';
export RepeatOne from './RepeatOne';
export Search from './Search';
export Sections from './Sections';
export Settings from './Settings';
export Share from './Share';
export Shuffle from './Shuffle';
export SkipForwardThirty from './SkipForwardThirty';
export SkipBackThirty from './SkipBackThirty';
export SkipNext from './SkipNext';
export SkipPrevious from './SkipPrevious';
export Text from './Text';
export Time from './Time';
export Umbrella from './Umbrella';
export Video from './Video';
export VideoOff from './VideoOff';
export Volume from './Volume';
export Warning from './Warning';
