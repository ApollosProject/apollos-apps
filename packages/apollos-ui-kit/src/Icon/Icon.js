import PropTypes from 'prop-types';
import React from 'react';
import { compose, pure, getContext } from 'recompose';
import { flow, camelCase, upperFirst, kebabCase } from 'lodash';

import { getIsLoading } from '../isLoading';
import Placeholder from '../Placeholder';

import * as uikitIcons from '../theme/icons';

const pascalCase = (string) =>
  flow(
    camelCase,
    upperFirst
  )(string);

// Convenience component to render icons based on the icon's string name, like:
// <Icon name="skip-next" {...allOtherPropsPassedToComponent} />
//
// Can also import the icon directly:
// import { SkipNext } from '../theme/icons';
// <SkipNext />

const enhance = compose(
  pure,
  getIsLoading,
  getContext({
    iconInput: PropTypes.objectOf(PropTypes.func),
  })
);

const Icon = enhance(
  ({ name, size, iconInput, isLoading = false, ...otherProps }) => {
    const Icons = { ...uikitIcons, ...iconInput };
    const IconComponent = Icons[pascalCase(name)];
    return (
      <Placeholder.Media size={size} hasRadius onReady={!isLoading}>
        <IconComponent size={size} {...otherProps} />
      </Placeholder.Media>
    );
  }
);

// eslint-disable-next-line consistent-return
const namePropValidator = (props, propName, componentName) => {
  const icons = Object.keys({ ...uikitIcons, ...props.iconInput }).map(
    kebabCase
  );

  if (!icons.includes(props.name)) {
    // eslint-disable-next-line no-console
    return new Error(
      `Invalid prop \`${propName}\` of value \`${
        props.name
      }\` supplied to \`${componentName}\` expected one of ${JSON.stringify(
        icons
      )}`
    );
  }
};

Icon.propTypes = {
  name: namePropValidator,
  size: PropTypes.number,
  fill: PropTypes.string,
  isLoading: PropTypes.bool,
};

Icon.defaultProps = {
  size: 32, // 32 is the default size used within the svg component
};

Icon.displayName = 'Icon';

export default Icon;
