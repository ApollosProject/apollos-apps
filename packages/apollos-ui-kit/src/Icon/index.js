import PropTypes from 'prop-types';
import React from 'react';
import { compose, pure } from 'recompose';
import { flow, camelCase, upperFirst, kebabCase } from 'lodash';

import Placeholder from '../Placeholder';

import * as Icons from './icons';

const pascalCase = (string) =>
  flow(
    camelCase,
    upperFirst
  )(string);

// Convenience component to render icons based on the icon's string name, like:
// <Icon name="skip-next" {...allOtherPropsPassedToComponent} />
//
// Can also import the icon directly:
// import { SkipNext } from 'Icon/icons';
// <SkipNext />

const enhance = compose(pure);

const Icon = enhance(({ name, size, isLoading = false, ...otherProps }) => {
  const IconComponent = Icons[pascalCase(name)];
  return (
    <Placeholder.Media size={size} hasRadius onReady={!isLoading}>
      <IconComponent size={size} {...otherProps} />
    </Placeholder.Media>
  );
});

Icon.propTypes = {
  name: PropTypes.oneOf(Object.keys(Icons).map(kebabCase)).isRequired,
  size: PropTypes.number,
  fill: PropTypes.string,
  isLoading: PropTypes.bool,
};

Icon.defaultProps = {
  size: 32, // 32 is the default size used within the svg component
};

export default Icon;
