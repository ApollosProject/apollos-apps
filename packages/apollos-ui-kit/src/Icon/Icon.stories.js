import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { kebabCase } from 'lodash';

import CenteredView from '../CenteredView';

import * as icons from './icons';

import Icon from '.';

const stories = storiesOf('Icon', module).addDecorator((getStory) => (
  <CenteredView>{getStory()}</CenteredView>
));

Object.keys(icons).forEach((iconName) => {
  stories.add(iconName, () => <Icon name={kebabCase(iconName)} />);
});

stories.add('isLoading', () => <Icon name={'umbrella'} isLoading />);
