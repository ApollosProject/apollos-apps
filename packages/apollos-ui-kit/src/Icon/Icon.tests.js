import React from 'react';
import renderer from 'react-test-renderer';
import { kebabCase } from 'lodash';

import Providers from '../Providers';

import * as icons from './icons';

import Icon from '.';

Object.keys(icons).forEach((iconName) => {
  describe(`The ${iconName} icon`, () => {
    it('renders correctly', () => {
      const tree = renderer.create(
        <Providers>
          <Icon name={kebabCase(iconName)} />
        </Providers>
      );
      expect(tree).toMatchSnapshot();
    });
  });
});
