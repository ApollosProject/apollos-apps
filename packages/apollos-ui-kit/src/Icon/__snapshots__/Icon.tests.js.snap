// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CoreIcons icon renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {},
    ]
  }
/>
`;

exports[`The Icon component accepts an iconInput object (custom icons) 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={40}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4278190080,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGPath
        d="M24.8 32h-18C5.2 32 4 30.8 4 29.3s1.2-2.8 2.8-2.8h18c.4 0 .8.3.8.8v4c-.1.4-.4.7-.8.7zm-18-4c-.7 0-1.3.6-1.3 1.3s.6 1.3 1.3 1.3H24V28H6.8z"
        fill={
          Array [
            0,
            4280756014,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
      <RNSVGPath
        d="M27.3 32h-2.5c-.4 0-.8-.3-.8-.8s.3-.8.8-.8h2.5c.4 0 .8.3.8.8s-.4.8-.8.8zM4.8 30.2l-.5-.2a.5.5 0 0 1-.3-.5v-.2c0-.4.3-.8.8-.8s.8.3.8.8v.1c0 .4-.2.7-.6.8h-.2z"
        fill={
          Array [
            0,
            4280756014,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
      <RNSVGPath
        d="M27.3 0H7.9A3.8 3.8 0 0 0 4 3.9v25.4c0 .4.3.7.8.7s.8-.3.8-.8c0-.7.6-1.3 1.3-1.3h20.5c.4 0 .8-.3.8-.8V.8c-.2-.5-.5-.8-.9-.8zM21 10.1h-4.3v12.8c0 .4-.3.8-.8.8s-.8-.3-.8-.8V10.1H11c-.4 0-.8-.3-.8-.8s.3-.8.8-.8h4.3V4.9c0-.4.3-.8.8-.8s.8.3.8.8v3.7H21c.4 0 .8.3.8.8s-.3.7-.8.7z"
        fill={
          Array [
            0,
            4280756014,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
]
`;

exports[`The Icon component overrides UIKit icons with iconInput (custom icons) that have the same name 1`] = `
Array [
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {},
      ]
    }
  />,
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={32}
    bbWidth={32}
    height={32}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        undefined,
        null,
        Object {
          "flex": 0,
          "height": 32,
          "width": 32,
        },
      ]
    }
    vbHeight={32}
    vbWidth={32}
    width={32}
  >
    <RNSVGGroup
      fill={
        Array [
          0,
          4278190080,
        ]
      }
      fillOpacity={1}
      fillRule={1}
      font={Object {}}
      matrix={
        Array [
          1,
          0,
          0,
          1,
          0,
          0,
        ]
      }
      opacity={1}
      originX={0}
      originY={0}
      propList={Array []}
      rotation={0}
      scaleX={1}
      scaleY={1}
      skewX={0}
      skewY={0}
      stroke={null}
      strokeDasharray={null}
      strokeDashoffset={null}
      strokeLinecap={0}
      strokeLinejoin={0}
      strokeMiterlimit={4}
      strokeOpacity={1}
      strokeWidth={1}
      vectorEffect={0}
      x={0}
      y={0}
    >
      <RNSVGPath
        d="M27 14h5c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2v0zM27 14c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2v0 14c0 1.112-0.895 2-2 2-1.112 0-2-0.896-2-2.001v-1.494c0-0.291 0.224-0.505 0.5-0.505 0.268 0 0.5 0.226 0.5 0.505v1.505c0 0.547 0.444 0.991 1 0.991 0.552 0 1-0.451 1-0.991v-14.009c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-1.105-1.119-2-2.5-2s-2.5 0.895-2.5 2c0-5.415 6.671-9.825 15-9.995v-1.506c0-0.283 0.224-0.499 0.5-0.499 0.268 0 0.5 0.224 0.5 0.499v1.506c8.329 0.17 15 4.58 15 9.995h-5z"
        fill={
          Array [
            0,
            4280756014,
          ]
        }
        fillOpacity={1}
        fillRule={1}
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
        opacity={1}
        originX={0}
        originY={0}
        propList={
          Array [
            "fill",
          ]
        }
        rotation={0}
        scaleX={1}
        scaleY={1}
        skewX={0}
        skewY={0}
        stroke={null}
        strokeDasharray={null}
        strokeDashoffset={null}
        strokeLinecap={0}
        strokeLinejoin={0}
        strokeMiterlimit={4}
        strokeOpacity={1}
        strokeWidth={1}
        vectorEffect={0}
        x={0}
        y={0}
      />
    </RNSVGGroup>
  </RNSVGSvgView>,
]
`;

exports[`The Icon component return null for a missing icon 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {},
    ]
  }
/>
`;

exports[`The PhosphorIcons icon renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {},
    ]
  }
/>
`;

exports[`The default icon renders correctly 1`] = `
<View
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      Object {},
    ]
  }
/>
`;
